src/main/java/com/google/cloud/tools/managedcloudsdk/ManagedCloudSdkFactory.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.managedcloudsdk;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/** {@link ManagedCloudSdk} factory. */
+public class ManagedCloudSdkFactory {
+
+  private Path managedSdkDirectory;
+  private OsInfo osInfo;
+
+  /** Create a new factory with hardcoded managed sdk root directory. */
+  public ManagedCloudSdkFactory() throws UnsupportedOsException {
```

Comment:
If the directory is always the same, should this be a singleton?

Reply:
The ManagedCloudSdkFactory?